### DISCUSSION TITLE
Sonoff Basic won't save wifi credential with version newer than 9.3.1

### DISCUSSION THREAD

**Discussion by frogale** [arendst/Tasmota#16052] [Q&A] [answered]
I have 3 Sonoff Basic (first edition) and with firmware newer than 9.3.1 I am not able to complete the initial setup.. It connects to my wifi for 1s and disconnects, LED flashes constantly, tasmota essid is no longer available. After restarting, the tasmota essid is present again and the wifi settings are not saved. When I use version 9.3.1 it works as expected and after upgrading to v12 it works as well. When I reset sonoff with v12 tasmota to defaults, I can't connect again and I have to downgrade to v9. Could this be a bug in the 3 main versions of Tasmota? Anyone with the same problem?

Edit: version 9.3.1 is the latest version with "old" wifi manager (with secondary essid)..
Edit2: Version 9.3.1 saves wifi credentials and reboots. Version 9.4.0+ tries to connect first and stucks if it can't.

**Comment by sfromis** [comment]
One clear risk factor is the crappiness of Sonoff Basic, as it *may* not be able to power the Wifi activity when connecting. You could try flashing the tasmota-lite binary, and see if it works better then.

**Reply by frogale** [reply]
Same problem with v12 lite version.. it seems to be a problem related to new wifi manager introduced on 9.4.0.. with old wifi manager on 9.3.1 it is ok...

**Reply by ascillato** [reply]
Hi, I can't reproduce your issue.

Let's find what is the problem in your case.

Please, when you connect to Tasmota AP, instead of going to http://192.168.4.1, please go to http://192.168.4.1/cs?
There you will access the Tasmota console. Please enter there `WEBLOG 4`, and then try the connection again.
After the attempt, please, check the console again and copy and paste the logs here.

thanks.

**Reply by frogale** [reply]
I'll try but as I wrote earlier after trying to connect the "tasmota" essid is gone so I can't connect to the console again without rebooting.. is it possible to access the console via serial interface or save the output to some permanent storage?

**Reply by frogale** [reply]
With serial console a got this:

00:00:00.001 HDW: ESP8266EX
00:00:00.026 CFG: Loaded from flash at FA, Count 2
00:00:00.032 QPC: Count 1
00:00:00.041 Project tasmota - Tasmota Version 12.0.2(tasmota)-2_7_4_9(2022-06-20T12:41:03)
00:00:00.199 WIF: WifiManager active for 3 minutes
00:00:00.707 HTP: Web server active on tasmota-4BB708-5896 with IP address 192.168.4.1
00:00:06.021 QPC: Reset
00:06:04.287 CMD: Seriallog 4
00:06:04.292 RSL: RESULT = {"SerialLog":{"4":{"Active":"4"}}}
00:06:04.728 CFG: Saved to flash at F4, Count 8, Bytes 4096
00:06:13.351 HTP: Configure WiFi
00:06:13.377 WIF: Scanning...
00:06:15.564 WIF: Scan done
00:06:29.379 HTP: Configure WiFi
00:06:29.381 CFG: CR 347/699, Busy 0
00:06:29.382 CFG: CR 355/699, Busy 0
00:06:29.383 WIF: Connecting to AP Wifinka2 as tasmota-4BB708-5896 ...
00:06:29.878 WIF: Trying to connect device to network Wifinka2
00:06:30.878 WIF: Trying to connect device to network Wifinka2
00:06:31.878 WIF: Trying to connect device to network Wifinka2
00:06:32.878 WIF: Trying to connect device to network Wifinka2
00:06:33.878 WIF: Trying to connect device to network Wifinka2
00:06:34.880 WIF: Trying to connect device to network Wifinka2
00:06:35.878 WIF: Trying to connect device to network Wifinka2
00:06:36.878 WIF: Trying to connect device to network Wifinka2
00:06:37.878 WIF: Trying to connect device to network Wifinka2
00:06:37.880 WIF: Connect failed as AP cannot be reached
00:06:41.174 HTP: Configure WiFi
00:06:41.200 WIF: Scanning...
00:06:43.388 WIF: Scan done

Nothing really interesting... But it should not be an issue with my wifi because with older version everything is working.. 

Edit: This is log from 9.3.1 version.. there is one restart between saving wifi settings and connection to my wifi:

00:00:00.049 CFG: Loaded from flash at FB, Count 12
00:00:00.142 QPC: Count 1
00:00:00.144 CFG: CR 331/699, Busy 0
00:00:00.147 SRC: Restart
00:00:00.149 Project tasmota Tasmota Version 9.3.1(tasmota)-2_7_4_9(2021-03-09T16:12:28)
00:00:00.401 WIF: Checking connection...
00:00:00.402 WIF: WifiManager active for 3 minutes
00:00:00.402 WIF: Wifimanager set AccessPoint
00:00:00.909 HTP: Web server active on tasmota_4BB708-5896 with IP address 192.168.4.1
00:00:07.098 QPC: Reset
00:00:09.041 APP: Boot Count 7
00:00:09.630 CFG: Saved to flash at FA, Count 14, Bytes 4096
00:00:30.506 HTP: Redirected to captive portal
00:00:30.555 HTP: Redirected to captive portal
00:00:31.670 HTP: Configure WiFi
00:00:56.648 HTP: Configure WiFi
00:00:58.201 HTP: Configure WiFi
00:01:00.409 WIF: Scan done
00:01:06.837 HTP: Redirected to captive portal
00:01:06.938 HTP: Redirected to captive portal
00:01:11.797 HTP: Configure WiFi
00:01:11.798 CFG: CR 331/699, Busy 0
00:01:11.799 CFG: CR 331/699, Busy 0
00:01:11.801 CFG: CR 331/699, Busy 0
00:01:11.802 CFG: CR 339/699, Busy 0
00:01:11.803 CFG: CR 339/699, Busy 0
00:01:11.807 CFG: CR 347/699, Busy 0
00:01:11.810 CFG: CR 347/699, Busy 0
00:01:11.813 WIF: Hostname %s-%04d, SSId1 Wifinka2, SSId2 , CORS
00:01:11.819 HTP: Restart
00:01:11.852 SRC: WebGui from 0.0.0.0
00:01:11.948 CFG: Saved to flash at F9, Count 15, Bytes 4096
00:01:12.789 APP: Restarting

 ets Jan  8 2013,rst cause:1, boot mode:(3,6)

load 0x4010f000, len 3584, room 16
tail 0
chksum 0xb0
csum 0xb0
v3969889e
~ld

00:00:00.049 CFG: Loaded from flash at F9, Count 15
00:00:00.055 QPC: Count 1
00:00:00.057 CFG: CR 347/699, Busy 0
00:00:00.061 SRC: Restart
00:00:00.062 Project tasmota Tasmota Version 9.3.1(tasmota)-2_7_4_9(2021-03-09T16:12:28)
00:00:00.214 WIF: Checking connection...
00:00:00.214 WIF: Attempting connection...
00:00:00.549 WIF: Connecting to AP1 Wifinka2 in mode 11n as tasmota_4BB708-5896...
00:00:01.407 WIF: Checking connection...
00:00:01.408 WIF: Attempting connection...
00:00:02.408 WIF: Checking connection...
00:00:02.409 WIF: Attempting connection...
00:00:04.001 WIF: Checking connection...
00:00:04.002 WIF: Attempting connection...
00:00:05.001 WIF: Checking connection...
00:00:05.003 WIF: Connected
00:00:05.253 HTP: Web server active on tasmota_4BB708-5896 with IP address 10.*anonymized*
00:00:05.653 WIF: Sending Gratuitous ARP
00:00:05.704 RTC: UTC 2022-07-24T12:24:28, DST 2022-03-27T02:00:00, STD 2022-10-30T03:00:00
13:24:28.139 CFG: Saved to flash at F8, Count 16, Bytes 4096
13:24:29.302 RSL: tele/tasmota_4BB708/INFO1 = {"Module":"Sonoff Basic","Version":"9.3.1(tasmota)","FallbackTopic":"cmnd/DVES_4BB708_fb/","GroupTopic":"cmnd/tasmotas/"}
13:24:29.308 RSL: tele/tasmota_4BB708/INFO2 = {"WebServerMode":"Admin","Hostname":"tasmota_4BB708-5896","IPAddress":"10.*anonymized*"}
13:24:29.320 RSL: tele/tasmota_4BB708/INFO3 = {"RestartReason":"Software/System restart"}
13:24:29.328 RSL: stat/tasmota_4BB708/RESULT = {"POWER":"OFF"}

**Reply by barbudor** [reply]
You wrote 'nothing interresting' but tasmota says:
> 00:06:37.880 WIF: Connect failed as AP cannot be reached

So you have problem connecting to your AP

What happens if you upgrade to recent version directly from 9.3.1? 
Using OTA from http://ota.tasmota.com directly to the device?

**Reply by frogale** [reply]
When upgrading from 9.3.1 directly to the latest 12.0 it is working.. until next reset to defaults..

**Reply by barbudor** [reply]
So I would bet your problem is about entering credentials, not about Tasmota (although you can still give a go to Jason's advice)

**Reply by frogale** [reply]
I really don't know where the problem is.. But version 9.3.1 is fine. If there is no one who can reproduce the same problem, it must be some compatibility issue between my Mikrotik router, Tasmota and Sonoff Basic hardware. But I can live with that (reset to defaults, downgrade, setup wifi, upgrade).

**Comment by lalo-uy** [comment]
Did you try to ser AP2 instead of AP1? That works for my some times.

**Reply by Jason2866** [reply]
Try: type `wifi 3` in a serial console to disable Wi-Fi 11n. Make sure that you have 11b or 11g enabled in your router.

**Reply by frogale** [reply]
> Did you try to ser AP2 instead of AP1? That works for my some times. El El sáb, 23 de jul. de 2022 a la(s) 15:46, Alex ***@***.***> escribió:
> […](#)
>

There is no AP2 option from 9.4.0 onwards.. you can set up only the first essid now..

**Reply by barbudor** [reply]
Wrong, there is still an AP2 (commands SSID2 and PASSWORD2) although I don't see how it could help

**Reply by frogale** [reply]
> Try: type `wifi 3` in a serial console to disable Wi-Fi 11n. Make sure that you have 11b or 11g enabled in your router.

No change with 11g mode.. "AP cannot be reached" again..

**Reply by barbudor** [reply]
Are you entering the ssid/password from the tasmota settings web page?
Is there any 'strange character' in either those?
Does it work if you set the ssid/password from the serial console? Or even better use the web installer at tasmota.github.io/install to do erase, flash and set wifi creds

**Reply by frogale** [reply]
> Wrong, there is still an AP2 (commands SSID2 and PASSWORD2) although I don't see how it could help

Ah.. I see.. you mean the password2 and ssid2 commands in the Tasmota console.. yes those commands (even password1 and ssid1) result in connecting to wifi.. but I already know v12 can connect to my router. The problem is with the "initial setup" after a reset to defaults or first flash. But it could be some kind of workaround for me.

**Reply by frogale** [reply]
> Are you entering the ssid/password from the tasmota settings web page? Is there any 'strange character' in either those? Does it work if you set the ssid/password from the serial console? Or even better use the web installer at tasmota.github.io/install to do erase, flash and set wifi creds

I mainly use install from web (with erase) but without entering ssid and password (I don't even know how).. next I connect to tasmota ssid with my phone.. go to IP 192.168.4.1 with my chrome mobile browser.. fill wifi ssid and key.. save.. nothing.. No special characters in ssid and password.. only basic "alphanum" is present.. When i connect to wifi from console (192.168.4.1/cs) it works.. looks like some kind of bug in the initial setup script but for now I am only one who have problems with it..

**Reply by barbudor** [reply]
Summary
- ssid/password work when upgrading without erasing
- ssid/password work when entered from the console
- ssid/password are only basic alphanum
- not a problem with the router (wifi 3 didn't changed a thing) 

What are the length of your ssid and password?
@ascillato2 did you enforced a minimum or a max length when you changed the wifi config portal or could there be a limit somehow?

**Reply by ascillato** [reply]
@barbudor 
> did you enforced a minimum or a max length when you changed the wifi config portal or could there be a limit somehow?

Nothing enforced.

I have also a Mikrotik and some sonoff basics and the AP Portal works fine.

@frogale

Do you have a 2.4G and a 5G network with the same SSID?

**Comment by ascillato** [comment, marked as answer]
@frogale

Hi,

Please, can you test again? This time, after setting the credentials and hitting save, wait and when Tasmota tells you that can't connect, hit save again and check if it could connect.

Tasmota on versions newer than 9.3.1 has a different system for the initial configuration. It actually test the credentials before saving them. What it does is to remain in the AP mode (in which your phone is connected) and also tries to connect to the router. The ESP8266 supports this dual mode of AP+STA. But that dual mode has a limitation that is not being able to connect to 11n-ONLY networks. So, to overcome that hardware limitation, on the second attempt that you hit save, Tasmota won't try to test the credentials and will save credentials into flash and restart as 9.3.1 does. It turns off the AP mode and leaves alone the STA mode which is the only hardware mode that can connect to WiFi Networks in Mode 11n-Only.

Please, try again and let us know if this solves your issue. Thanks.
